htscodecs/tokenise_name3.c

```diff
@@ -1241,58 +1241,138 @@ static int64_t rans_decode(uint8_t *in, uint64_t in_len, uint8_t *out, uint64_t
     return clen+nb;
 }
 
-static int compress(uint8_t *in, uint64_t in_len, int level, int use_arith,
+static int compress(uint8_t *in, uint64_t in_len, enum name_type type,
+                    int level, int use_arith,
                     uint8_t *out, uint64_t *out_len) {
     uint64_t best_sz = UINT64_MAX;
-    int best = 0;
     uint64_t olen = *out_len;
 
-    //fprintf(stderr, "=== try %d ===\n", (int)in_len);
-
-    int m, rmethods[5][12] = {
-        {2,   0,      128},                                   // 1
-        {2,   0,                         192+8},              // 3
-        {3,   0,  128,                   193+8},              // 5
-        {6,   0,1,    129,   65,    193, 193+8},              // 7
-        {9,   0,1,128,129,64,65,192,193, 193+8},              // 9
-    };
-
-    // 1-9 to 0-4
+    // Map levels 1-9 to 0-4, for parameter lookup in R[] below
     level = (level-1)/2;
     if (level<0) level=0;
     if (level>4) level=4;
 
-    for (m = 1; m <= rmethods[level][0]; m++) {
+    // rANS4x16pr and arith_dynamic parameters to explore.
+    // We brute force these, so fast levels test 1 setting and slow test more
+    int R[5][16][7] = {
+        {   // -1
+            /* TYPE         */ {1, 128},
+            /* ALPHA        */ {1, 129},
+            /* CHAR         */ {1, 0},
+            /* DIGITS0      */ {1, 8},
+            /* DZLEN        */ {1, 0},
+            /* DUP          */ {1, 8},
+            /* DIFF         */ {1, 8},
+            /* DIGITS       */ {1, 8},
+            /* DDELTA       */ {1, 0},
+            /* DDELTA0      */ {1, 128},
+            {1,0},{1,0},{1,0},{1,0},{1,0}
```

Comment:
You're one `{1,0}` short on all these initialiser-lists.  I'm not sure if it's possible to hit the missing entry, but if it does I think something bad would happen as you'd not try any codecs, and might then attempt to copy `UINT64_MAX` bytes from an uninitialised buffer to the output.